### ISSUE TITLE
Tag Preview patch crashes when using mulitple monitors

### ISSUE THREAD

**Issue by Zaedus** [bakkeby/dwm-flexipatch#308]
When I switch tags after occupying another tag with a client, dwm crashes. I've been having this issue for awhile and just recently decided to take a look into fixing it since I don't usually use multiple monitors. After a little bit of debugging I found that the line of code found [here in dwm.c on line 4878](https://github.com/bakkeby/dwm-flexipatch/blob/master/dwm.c#L4878) causes a crash by calling `tagpreviewswitchtag()`. This error only occurs when there are multiple bars/monitors if that's any help to someone who wants to investigate this. For anyone experiencing this issue, disabling the tag preview patch fixes the issue at the cost of cool tag previews.

**Comment by bakkeby** [comment]
> I don't usually use multiple monitors

That is actually a bigger clue than you may realise.

I imagine that you may be starting dwm on a laptop with a single screen, then you hotplug a second screen.

In this scenario the reason it will crash is because the preview window does not exist for the second monitor. The reason for this is that the preview window is only created in the `updatepreview` function - and a call to that only happens once in the `setup` function. This is also an issue in the original [patch](https://dwm.suckless.org/patches/tag-previews/).

I'd expect that the same issue can be reproduced by removing a monitor and adding it again using `xrandr` / `arandr`.

I refactored this quite a bit for one of my other projects. I'll see about adding a fix for this issue, thanks for reporting.

**Comment by bakkeby** [comment]
I pushed a proposed fix for the above issue.

Another issue I found was that if you use `xrandr` / `arandr` to change the resolution of a monitor then the size of the preview window does not scale accordingly (i.e. it may be very large or very small depending on whether the resolution is increased or decreased, rather than being 1/4th of the monitor size as per config). This should be fixed as well.

Let me know if you have any further issues.
